niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/SignaturePermissionTester.java

```diff
@@ -3656,18 +3875,625 @@ public void writeToParcel(Parcel parcel, int i) {
+                                        return Consumer.super.andThen(after);
+                                    }
+                                });
+                    } catch(UnexpectedPermissionTestFailureException e){
+                        //The error is intended
+                        //Lost mode location updates can only be sent on an organization-owned device.
+                    }
+                }));
+
+        // # Skip START_CROSS_PROFILE_ACTIVITIES
+        // Reason : CrossProfileAppService requires Android Enterprise Environment.
+
+        mPermissionTasks.put(permission.QUERY_USERS,
+                new PermissionTest(false, Build.VERSION_CODES.TIRAMISU, () -> {
+                    invokeReflectionCall(mUserManager.getClass(),
+                            "getUserRestrictionSource", mUserManager,
+                            new Class<?>[]{java.lang.String.class, android.os.UserHandle.class},
+                            "Hello",UserHandle.getUserHandleForUid(0));
+                }));
+
+        mPermissionTasks.put(permission.QUERY_ADMIN_POLICY,
+                new PermissionTest(false, Build.VERSION_CODES.TIRAMISU, () -> {
+                    invokeReflectionCall(mDevicePolicyManager.getClass(),
+                            "getWifiSsidPolicy", mDevicePolicyManager,
+                            new Class<?>[]{});
+                }));
+
+        mPermissionTasks.put(permission.PROVISION_DEMO_DEVICE,
+            new PermissionTest(false, Build.VERSION_CODES.TIRAMISU, () -> {
+                Class<?> fmdpBuilderClazz = null;
+                Class<?> fmdpClazz = null;
+                try {
+                    fmdpBuilderClazz = Class.forName("android.app.admin.FullyManagedDeviceProvisioningParams$Builder");
+                    fmdpClazz = Class.forName("android.app.admin.FullyManagedDeviceProvisioningParams");
+
+                    Constructor constructor = fmdpBuilderClazz.getConstructor(ComponentName.class,String.class);
+                    Object fmdpBuilderObj =
+                        constructor.newInstance(new ComponentName(mContext,MainActivity.class),"");
+                    Object fmdpObj = invokeReflectionCall(fmdpBuilderClazz,
+                            "build", fmdpBuilderObj,
+                            new Class<?>[]{});
+                    invokeReflectionCall(mDevicePolicyManager.getClass(),
+                            "provisionFullyManagedDevice", mDevicePolicyManager,
+                            new Class<?>[]{fmdpClazz},fmdpObj);
+                } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException e) {
+                    throw new UnexpectedPermissionTestFailureException(e);
+                } catch (UnexpectedPermissionTestFailureException e){
+                    mLogger.logDebug("Please check stack trace. " +
```

Comment:
Is this something we could potentially check for in a future commit? Doesn't have to be part of the Android T pull request, but maybe as a follow-on commit to since it is likely few will see this message if it doesn't fail the test.

Reply:
No,  if the test 'succeed' this message will always show up. (if we run the test without platform signature we'll get a secuirty exception)

1. The test execute PolicyManager.provisionFullyManagedDevice method to verify "PROVISION_DEMO_DEVICE" perm
2. If we succeed to call the method always fail because of the "android.os.ServiceSpecificException"
3. But I cannot catch this exception because that is wrapped into UnexpectedPermissionTestFailureException and InvocationTargetException. 

As you know InvocationTargetException is shown if we call a hidden method with wrong parameter or signatures.

Let me think a while, if I can pick up this exception from the nested stacktrace.

Reply:
Modified the test to evaluate the nested exception correctly. Fix.

Reply:
Change looks good to me; thanks!